plugins/Web/plugin.py

```diff
@@ -149,38 +149,50 @@ def noIgnore(self, irc, msg):
     def getTitle(self, irc, url, raiseErrors):
         size = conf.supybot.protocols.http.peekSize()
         timeout = self.registryValue('timeout')
-        (target, text) = utils.web.getUrlTargetAndContent(url, size=size,
-                timeout=timeout)
-        try:
-            text = text.decode(utils.web.getEncoding(text) or 'utf8',
-                    'replace')
-        except UnicodeDecodeError:
-            pass
-        if minisix.PY3 and isinstance(text, bytes):
-            if raiseErrors:
-                irc.error(_('Could not guess the page\'s encoding. (Try '
-                        'installing python-charade.)'), Raise=True)
-            else:
-                return None
         try:
-            parser = Title()
-            parser.feed(text)
-        except UnicodeDecodeError:
-            # Workaround for Python 2
-            # https://github.com/ProgVal/Limnoria/issues/1359
-            parser = Title()
-            parser.feed(text.encode('utf8'))
-        parser.close()
-        title = utils.str.normalizeWhitespace(''.join(parser.data).strip())
-        if title:
-            return (target, title)
-        elif raiseErrors:
-            if len(text) < size:
-                irc.error(_('That URL appears to have no HTML title.'),
-                        Raise=True)
-            else:
-                irc.error(format(_('That URL appears to have no HTML title '
-                                 'within the first %S.'), size), Raise=True)
+            (target, text) = utils.web.getUrlTargetAndContent(url, size=size,
+                timeout=timeout)
+        except Exception as e:
+            self.log.info('Web plugin TitleSnarfer: URL <%s> raised <%s>', url, str(e))
+            text = None
+        if text:
+            try:
+                text = text.decode(utils.web.getEncoding(text) or 'utf8',
+                        'replace')
+            except UnicodeDecodeError:
+                pass
+            if minisix.PY3 and isinstance(text, bytes):
+                if raiseErrors:
+                    irc.error(_('Could not guess the page\'s encoding. (Try '
+                            'installing python-charade.)'), Raise=True)
+                else:
+                    return None
+            try:
+                parser = Title()
+                parser.feed(text)
+            except UnicodeDecodeError:
+                # Workaround for Python 2
+                # https://github.com/ProgVal/Limnoria/issues/1359
+                parser = Title()
+                parser.feed(bytes(text)) # bytes() = str() in py2
```

Comment:
That's re-encoding a decoded string. You could use the original value of `test` instead (ie. the one before `text.decode`)

Reply:
I changed this from 'parser.feed(text.encode('utf8'))' that was also re encoding things. Not sure but after reading about py2 it seems that there's no difference between this change or how it's actually, so maybe just leave it alone? I'm really unsure about this one, just know the first change fixed things for a unique URL as discussed in #1359. I can perform like you suggest but im far away from understading the logic, maybe changing like you suggest we will return to the situation before #1359? I'm really lost with this one.

Reply:
This line decodes the text, ie turns it from str to bytes, in a lossy way: https://github.com/ProgVal/Limnoria/pull/1377/files#diff-9a20c614ecc345c998e528d2a13f1686R160
And doing `bytes()` on the result does the same thing the other way around.

So it's better to use the original text

Reply:
Except we're dealing with py2 instead of py3 here, and I forgot how it works. * shrug *

Note that I'll start removing py2 support from Limnoria in 2020

Reply:
I just returned to the original value (before my commits)